plugins/inputs/gnmi/handler.go

```diff
@@ -98,21 +114,55 @@ func (h *handler) subscribeGNMI(ctx context.Context, acc telegraf.Accumulator, t
 				h.log.Debugf("update_%v: %s", t, string(buf))
 			}
 		}
-
 		if response, ok := reply.Response.(*gnmiLib.SubscribeResponse_Update); ok {
-			h.handleSubscribeResponseUpdate(acc, response)
+			h.handleSubscribeResponseUpdate(acc, response, reply.GetExtension())
 		}
 	}
 	return nil
 }
 
 // Handle SubscribeResponse_Update message from gNMI and parse contained telemetry data
-func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, response *gnmiLib.SubscribeResponse_Update) {
+func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, response *gnmiLib.SubscribeResponse_Update, extension []*gnmiExt.Extension) {
 	var prefix, prefixAliasPath string
 	grouper := metric.NewSeriesGrouper()
 	timestamp := time.Unix(0, response.Update.Timestamp)
 	prefixTags := make(map[string]string)
 
+	// check if there are exceptions
+	if len(extension) != 0 {
+		// iter on each extension
+		for _, ext := range extension {
+			currentExt := ext.GetRegisteredExt().Msg
+			if currentExt == nil {
+				break
+			}
+			// extension ID
+			switch ext.GetRegisteredExt().Id {
+			// Juniper Telemetry header
+			//EID_JUNIPER_TELEMETRY_HEADER = 1;
+			case 1:
+				// Decode it only if user requested it
+				if choice.Contains("juniper_header", h.vendorExt) {
+					juniperHeader := &jnprHeader.GnmiJuniperTelemetryHeaderExtension{}
+					// unmarshal extention
+					err := proto.Unmarshal(currentExt, juniperHeader)
+					if err != nil {
+						h.log.Errorf("unmarshal gnmi Juniper Header extention failed: %v", err)
```

Comment:
Please wrap errors:
```suggestion
						h.log.Errorf("unmarshal gnmi Juniper Header extention failed: %w", err)
```
Also, at some places you named it "Juniper Extension Header" versus here "Juniper Header extension". Align all references would be preferred.

Reply:
change done. Aligned all to  Juniper Header Extension